### ISSUE TITLE
Receiving from P2SH output

### ISSUE THREAD

**Issue by robjohnson189** [mycelium-com/wallet#139]
When a transaction to mycelium is unconfirmed and it is using a P2SH output for an input, it doesn't show up as "Receiving" in mycelium. It does show up, after the transaction is confirmed.

**Comment by jandreske** [comment]
At the moment, our servers use a custom transaction validation mechanism, and only ask the full node for confirmed transactions. That way, we can learn a lot more about the transactions while validating them, like estimating how likely it is they get double spent by a mutated version etc. This mechanism simply did not have a validation implemented for P2SH so far, thats why such transactions only showed up after confirmation. I already implemented validation for P2SH multisig, thats probably the biggest application of P2SH right now. Payments from P2SH multisig should be shown as receiving after we tested everything and updated the servers. Other P2SH stuff will probably be implemented later, when usage grows.
